GitCommands/Settings/ConfigFileSettings.cs

```diff
@@ -70,16 +70,26 @@ public static ConfigFileSettings CreateGlobal(ConfigFileSettings? lowerPriority,
                 SettingLevel.SystemWide);
         }
 
-        public new string GetValue(string setting)
-        {
-            return GetString(setting, string.Empty);
-        }
+        public new string GetValue(string setting) => GetString(setting, string.Empty);
+
+        public T? GetValue<T>(string setting) where T : struct => ConvertValue<T>(GetValue(setting), setting);
 
-        public IReadOnlyList<string> GetValues(string setting)
+        private T? ConvertValue<T>(string value, string setting) where T : struct
```

Comment:
I would not make this nullable, instead the caller has to declare `T` or `T?` as that makes the contract explicit.

Reply:
We are a little constraint here by the `Convert.ChangeType()` method that doesn't accept nullable. So we will have to extract the underlying type and it's a bit strange to do.

And for me the `null` was making sens for me to handle the 2 cases where value is not of the type expected and more important, the setting is not set. With a nullable as a return type it's more explicit.

And also I find the the generic constraint on `struct` more dev friendly on the type we can expect to convert.